spec/support/delayed_job.rb

```diff
@@ -87,6 +82,55 @@ def query_jobs(object, method_name)
     end
   end
 
+  # Usage:
+  #    expect(Mailer).to delay_email_delivery_of(:some_email).with(arguments / argument_matchers)
+  #
+  # Examples:
+  #    Mailer.email(name: "My name").deliver_later
+  #
+  #    expect(Mailer).to delay_email_delivery_of(:email).with(hash_including :name) #=> success
+  #    expect(Object).to delay_email_delivery_of(:email)                            #=> success
+  #    expect(Object).to delay_email_delivery_of(:email).with(name: "My name")      #=> success
+  #    expect(Object).to delay_email_delivery_of(:email).with(name: "Hello")        #=> failure
+  #
+  # How:
+  #   `deliver_later` comes from ActiveJob, which wraps the mailer call inside `ActionMailer::MailDeliveryJob`,
+  #   which inherits from `ActiveJob::Base`. This job, as all ActiveJob::Base jobs, is then wrapped in
+  #   yet another wrapper for Delayed::Job - ActiveJob::QueueAdapters::DelayedJobAdapter::JobWrapper.
+  #
+  #   To effectively query and access arguments passed to the mailer we need to unwrap both layers
+
+  define :delay_email_delivery_of do |email_name|
+    match do |mailer_class|
+      email_arguments = find_email_args(mailer_class, email_name)
+      return email_arguments.any? unless @arguments
+
+      email_arguments.any? do |args|
+        @arguments.args_match?(*args)
+      end
+    end
+
+    chain :with do |*args|
+      @arguments = RSpec::Mocks::ArgumentListMatcher.new(*args)
+    end
```

Comment:
Is it worth adding an :at matcher here?

Reply:
I think that would be the point to apply YAGNI. There is no place that chain would be used now and I don't think any requirements for that are coming. :) Actually, I don't think I have ever had to schedule email delivery on its own.